spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+  describe 'finds a searched user by name' do
+    #prepare
+    let!(:user) { create(:user, name: 'Dan') }
+    let!(:user1) { create(:user, name: 'Edimo') }
+    let!(:user2) { create(:user, name: 'Edmilson') }
+
+    it 'finds by name Ed' do
+      #action
+      get :index, params: { search_by_name: 'Ed' }
+      #assert
+      assigns(:users).should eq([user1, user2])
+    end
+
+    it 'finds by name with da' do
+      #action
+      get :index, params: { search_by_name: 'Da' }
+      #assert
+      assigns(:users).should eq([user])
+    end
+
+    it 'when does not find any user' do
+      #action
```

Comment:
it is good to create users here even if you know that they are not coming. Think like this, if some issue happens and the method returns all users when you use a query that supposed to return an empty array, this test will not catch it

Reply:
it is already done